### PR TITLE
Add TypeScript typings

### PR DESCRIPTION
**Pull request by noveogroup-amorgunov**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+declare module 'wheel-indicator' {
+    type WheelIndicatorEvent = Event & {
+        direction: string;
```
Comment (f0rmat1k) [inline index.d.ts]: `direction` should be `'up' | 'down'`

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        callback?: (event: WheelIndicatorEvent) => void;
+        preventMouse?: boolean;
+    }
+
+    export default class WheelIndicator {
+        constructor(options: Options);
+
+        destroy(): WheelIndicator;
+
+        getOption(option: keyof Options): any;
```
Comment (f0rmat1k) [inline index.d.ts]: I don't know all features of typescript well, but i believe here should be some generic, that declares, that returnable type depends on option. I found this solution, but not sure that it's right:
```
type ValueOf<T> = T[keyof T];
...
getOption(option: keyof Options): ValueOf<Options>;
```

File: lib/wheel-indicator.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /**
  * Generates event when user makes new movement (like a swipe on a touchscreen).
- * @version 1.2.2
+ * @version 1.2.3
```
Comment (f0rmat1k) [inline lib/wheel-indicator.js]: we should remove this precommit hook)

**Review by f0rmat1k — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (ulitcos) [reply index.d.ts]: looks like a working solution
https://www.typescriptlang.org/play/index.html?ssl=16&ssc=1&pln=45&pc=1#code/C4TwDgpgBAaghgGwK4QPIDMA8AVAfFAXimwG0BrCEAe3WIF0BuAKCYEsA7YCAJ3TgGNoASXYBnYHHaCoAbyhMoiqOzgBbCAC4o47hwDmzJVDh7NypKoBGPZgF8W-KmOBQO4yYK0j3U6ERkKSirqWgDkcKEANIGKJmYAjABMTPZM6EhSwKxOUKbAqGBZTgAUVIXZ7FoU1LTeEr4AlFrwyGhYdR4Q+HIxUNwQwEjc7K7OnSRlRex0KUxAA

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+declare module 'wheel-indicator' {
+    type ValueOf<T> = T[keyof T];
+
+    type WheelIndicatorEvent = Event & {
```
Comment (f0rmat1k) [inline index.d.ts]: missed past time. `WheelEvent` instead of `Event`

**Review by f0rmat1k — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 4 above)
Comment (noveogroup-amorgunov) [reply index.d.ts]: What do you mean? Callback (https://github.com/Promo/wheel-indicator/blob/master/lib/wheel-indicator.js#L28) get a browser event, with custom prop direction. So, `WheelIndicatorEvent` should expand Event type.

Comment (f0rmat1k) [reply index.d.ts]: В колбэк приходит не `Event`, а `WheelEvent` https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent

Тут нужно расширять его:
`type WheelIndicatorEvent = WheelEvent & {`

**Review by f0rmat1k — APPROVED** [review]

**Comment by f0rmat1k** [comment]
wheel-indicator@1.3.0

**Pull request merged**